circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/eccvm/eccvm_circuit_builder.hpp

```diff
@@ -0,0 +1,524 @@
+
+        rows.z_perm_shift = typename Flavor::Polynomial(rows.z_perm.shifted());
+
+        const auto evaluate_relation = [&]<typename Relation>(const std::string& relation_name) {
+            auto relation = Relation();
+            typename Relation::RelationValues result;
+            for (auto& r : result) {
+                r = 0;
+            }
+            constexpr size_t NUM_SUBRELATIONS = result.size();
+
+            for (size_t i = 0; i < num_rows; ++i) {
+                typename Flavor::RowPolynomials row;
+                for (size_t j = 0; j < NUM_POLYNOMIALS; ++j) {
+                    row[j] = rows[j][i];
+                }
+                relation.add_full_relation_value_contribution(result, row, params, 1);
+
+                bool x = true;
+                for (size_t j = 0; j < NUM_SUBRELATIONS; ++j) {
+                    if (result[j] != 0) {
+                        info("Relation ", relation_name, ", subrelation index ", j, " failed at row ", i);
+                        x = false;
+                    }
+                }
+                if (!x) {
+                    return false;
+                }
+            }
+            return true;
+        };
+
+        bool result = true;
+        result = result && evaluate_relation.template operator()<honk::sumcheck::ECCVMTranscriptRelation<FF>>(
+                               "ECCVMTranscriptRelation");
+        result = result && evaluate_relation.template operator()<honk::sumcheck::ECCVMPointTableRelation<FF>>(
+                               "ECCVMPointTableRelation");
+        result =
+            result && evaluate_relation.template operator()<honk::sumcheck::ECCVMWnafRelation<FF>>("ECCVMWnafRelation");
+        result =
+            result && evaluate_relation.template operator()<honk::sumcheck::ECCVMMSMRelation<FF>>("ECCVMMSMRelation");
+        result =
+            result && evaluate_relation.template operator()<honk::sumcheck::ECCVMSetRelation<FF>>("ECCVMSetRelation");
+
+        auto lookup_relation = honk::sumcheck::ECCVMLookupRelation<FF>();
+        typename honk::sumcheck::ECCVMLookupRelation<typename Flavor::FF>::RelationValues lookup_result;
+        for (auto& r : lookup_result) {
+            r = 0;
+        }
+        for (size_t i = 0; i < num_rows; ++i) {
+            typename Flavor::RowPolynomials row;
+            for (size_t j = 0; j < NUM_POLYNOMIALS; ++j) {
+                row[j] = rows[j][i];
+            }
+            {
+                lookup_relation.add_full_relation_value_contribution(lookup_result, row, params, 1);
+            }
+        }
+        for (auto r : lookup_result) {
+            if (r != 0) {
+                info("Relation ECCVMLookupRelation failed.");
+                return false;
+            }
+        }
+        return result;
+    }
+
+    [[nodiscard]] size_t get_num_gates() const
+    {
+        // TODO(@zac-williamson) once we have a stable base to work off of, optimise this method!
```

Comment:
Could you create an issue?

Reply:
✅ [#2218](https://github.com/AztecProtocol/aztec-packages/issues/2218)